sklearn/cluster/tests/test_hdbscan.py

```diff
@@ -78,6 +78,17 @@ def test_hdbscan_distance_matrix():
     score = fowlkes_mallows_score(y, labels)
     assert score >= 0.98
 
+    msg = r"The precomputed distance matrix.*has shape"
+    with pytest.raises(ValueError, match=msg):
+        HDBSCAN(metric="precomputed", copy=True).fit_predict(X)
+
+    msg = r"The precomputed distance matrix.*values"
+    # Ensure the matrix is not symmetric
+    D[0, 1] = 10
+    D[1, 0] = 1
+    with pytest.raises(ValueError, match=msg):
+        HDBSCAN(metric="precomputed").fit_predict(D)
```

Comment:
Currently there is a [coverage warning because the case of `D` being a sparse non-CSR distance matrix is not tested](https://github.com/scikit-learn/scikit-learn/pull/24701/checks?check_run_id=11103907197).

Can you parametrize this test so that the case of `D` being a sparse non-CSR matrix is tested?